Lib/asyncio/base_tasks.py

```diff
@@ -74,3 +76,51 @@ def _task_print_stack(task, limit, file):
     if exc is not None:
         for line in traceback.format_exception_only(exc.__class__, exc):
             print(line, file=file, end='')
+
+
+# WeakKeyDictionary of {Task: EventLoop} containing all tasks alive.
+_all_tasks = weakref.WeakKeyDictionary()
```

Comment:
Why `weakref.WeakKeyDictionary` and not `weakref.WeakValueDictionary`?

Reply:
A task should be weak, on task garbage collection an item from `_all_tasks` should disappear. `asyncio` on master branch uses `WeakSet` of tasks.
But `task._loop` is a private property, custom task implementation can not expose it. That's why storing a loop along with task itself is required.

Mapping like `{loop: WeakSet[task]}` is not an option: there is no signal for loop deletion and `_all_tasks` is the singleton. In case of many loops creations/deletions (unit tests) the singleton will blow up with closed loops. I want to avoid it.

Reply:
Alright. This deserves to be put in a comment.

Reply:
ok